src/common/api/common/utils/EntityUpdateUtils.ts

```diff
@@ -31,37 +30,37 @@ export enum PrefetchStatus {
 	NotAvailable = "NotAvailable", // 403 (not authorized), 404 (not found)
 }
 
-export async function entityUpdateToUpdateData(
+export async function entityUpdateToUpdateData<T extends SomeEntity>(
 	clientTypeModelResolver: ClientTypeModelResolver,
 	update: EntityUpdate,
 	instance: Nullable<ServerModelParsedInstance> = null,
 	prefetchStatus: PrefetchStatus = PrefetchStatus.NotPrefetched,
-): Promise<EntityUpdateData> {
+): Promise<EntityUpdateData<T>> {
 	const typeId = update.typeId ? parseInt(update.typeId) : null
 	const typeIdOfEntityUpdateType = typeId
 		? new TypeRef<SomeEntity>(update.application as AppName, typeId)
 		: clientTypeModelResolver.resolveTypeRefFromAppAndTypeNameLegacy(update.application as AppName, update.type)
 
 	return {
 		typeRef: typeIdOfEntityUpdateType,
-		instanceListId: update.instanceListId,
+		instanceListId: update.instanceListId === "" ? null : update.instanceListId,
 		instanceId: update.instanceId,
 		operation: update.operation as OperationType,
 		patches: update.patch?.patches ?? null,
 		instance,
 		prefetchStatus,
-	}
+	} as EntityUpdateData<T>
```

Comment:
let's just cast `update.instanceListId` instead of the whole thing. Casting the whole object is a bit dangerous if we add a new field